src/packages/BooleanQueryTokenizer/BooleanQueryTokenizer.ts

```diff
@@ -148,6 +155,94 @@ export class BooleanQueryTokenizer {
         );
     }
 
+    public static extractWords(inputString: string): Extract {
```

Comment:
I don't know if it was discussed, but for now we just ignore invalid negations like
```
java AND NOT (js OR python
```
now it will return: `["java", "js", "python"]`

And this is fine because we don't know what exactly user wanted to do and where he made a mistake.
Just I think we need to mention in doc that we ignore invalid negations